lib/msal-common/src/cache/CacheManager.ts

```diff
@@ -204,23 +214,103 @@ export abstract class CacheManager implements ICacheManager {
         if (accountEntities.length < 1) {
             return [];
         } else {
-            const allAccounts = accountEntities.map<AccountInfo>((accountEntity) => {
+            const allAccounts: AccountInfo[] = accountEntities.map<AccountInfo>((accountEntity) => {
                 return this.getAccountInfoFromEntity(accountEntity);
             });
             return allAccounts;
         }
     }
+    
+    /**
+     * Returns all accounts with their respective tenant profiles in the cache
+     */
+    private getAllAccountsMultiTenant(): AccountInfo[] {
+        const allAccountKeys = this.getAccountKeys();
+        const allAccounts: AccountInfo[] = this.getAllAccounts();
```

Comment:
This will cause an infinite loop getAllAccounts -> getAllAccountsMultiTenant -> getAllAccounts -> ...

Reply:
I have no idea why this didn't break my sample but thanks for catching it! I probably missed it while refactoring. It was probably going to show up when writing the tests ofc.

Reply:
Ok the loop wasn't infinite because the getAllAccounts call here doesn't have the multinenantAccountsEnabled parameter set to true. It did loop once though, fixing by changing to this.getAllCachedAccounts instead, which was originally getAllAccounts.